src/components/d-active-card/DActiveCard.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { ImageBackground, Text, TouchableOpacity, View } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+
+import { styles } from './dActiveCard-styles'
+
+import { Challenge } from '../../screens/d-active-challenges/DActiveChallenges'
+
+import { SPACING } from '../../theme/spacing'
+import { backgroundGradients } from '../../theme/color'
+
+import { buttonsData } from '../../constants/dActivechallenges-constants'
+import { ResizeMode } from '../../constants/common-constants'
+import { getImage } from '../../utils/common-utils'
+
+/** DActiveCard: {This function shows active cards according to the data given.} */
+const DActiveCard = ({ data }: { data: Challenge }) => {
+  const renderButtons = () => {
+    return buttonsData.map((time, index) => (
+      <TouchableOpacity style={styles.btn} key={index}>
+        <Text style={styles.btntxt}>{time}</Text>
+      </TouchableOpacity>
+    ))
+  }
+  return (
+    <View style={styles.container}>
+      <ImageBackground
+        source={getImage(data.name)}
+        resizeMode={ResizeMode.Contain}
+        style={[styles.background, { elevation: data.dropShadow || data.cardShadow || 0 }]}>
```

Comment:
destructure data on top of the component

Reply:
I tried it but destructing data on top does not give me the data and no errors comes.